src/H5Fefc.c

```diff
@@ -231,54 +272,46 @@ H5F__efc_open(H5F_efc_t *efc, const char *name, unsigned flags, hid_t fcpl_id, h
              * do not add it to cache */
             if (ent) {
                 if (H5F__efc_remove_ent(efc, ent) < 0)
-                    HGOTO_ERROR(H5E_FILE, H5E_CANTREMOVE, NULL,
+                    HGOTO_ERROR(H5E_FILE, H5E_CANTREMOVE, FAIL,
                                 "can't remove entry from external file cache");
 
                 /* Do not free ent, we will recycle it below */
             } /* end if */
             else {
-                /* Cannot cache file, just open file and return */
-                if (NULL == (ret_value = H5F_open(name, flags, fcpl_id, fapl_id)))
-                    HGOTO_ERROR(H5E_FILE, H5E_CANTOPENFILE, NULL, "can't open file");
+                /* Cannot cache file, just try opening file and return */
+                if (H5F__efc_try_open_file(file, name, flags, fcpl_id, fapl_id) < 0)
+                    HGOTO_ERROR(H5E_FILE, H5E_CANTOPENFILE, FAIL, "can't try opening file");
 
-                /* Make file post open call */
-                if (H5F__post_open(ret_value) < 0)
-                    HGOTO_ERROR(H5E_FILE, H5E_CANTINIT, NULL, "can't finish opening file");
-
-                /* Increment the number of open objects to prevent the file from
-                 * being closed out from under us - "simulate" having an open
-                 * file id */
-                ret_value->nopen_objs++;
-
-                HGOTO_DONE(ret_value);
+                HGOTO_DONE(SUCCEED);
             } /* end else */
         }     /* end if */
         else
             /* Allocate new entry */
             if (NULL == (ent = H5FL_MALLOC(H5F_efc_ent_t)))
-                HGOTO_ERROR(H5E_RESOURCE, H5E_NOSPACE, NULL, "memory allocation failed");
+                HGOTO_ERROR(H5E_FILE, H5E_CANTALLOC, FAIL, "memory allocation failed");
```

Comment:
Question: why is H5E_FILE instead of H5E_RESOURCE in this case?

Reply:
Because the error occurred in a file module, not in a memory manager (i.e. "resource") module.